core/os/os.cpp

```diff
@@ -203,16 +203,20 @@ uint64_t OS::get_embedded_pck_offset() const {
 }
 
 // Helper function to ensure that a dir name/path will be valid on the OS
-String OS::get_safe_dir_name(const String &p_dir_name, bool p_allow_dir_separator) const {
+String OS::get_safe_dir_name(const String &p_dir_name, bool p_allow_paths) const {
+	String safe_dir_name = p_dir_name;
 	Vector<String> invalid_chars = String(": * ? \" < > |").split(" ");
-	if (p_allow_dir_separator) {
+	if (p_allow_paths) {
 		// Dir separators are allowed, but disallow ".." to avoid going up the filesystem
 		invalid_chars.push_back("..");
+		safe_dir_name = safe_dir_name.replace("\\", "/").strip_edges();
 	} else {
 		invalid_chars.push_back("/");
+		invalid_chars.push_back(".");
```

Comment:
Why do we need to forbid dots? Seems uqite restrictive.

Reply:
This is for the userdata folder, which is not visible to the user either way. It avoids folders like "." or "..".

Reply:
What if uses want to create `user://Jetpaca 2.0 Logs/`?

Reply:
I think its pointless, but if it makes you happy I can change the check